jobs/pbm_helpers.go

```diff
@@ -0,0 +1,377 @@
+	cmdTimeout          = time.Minute
+	resyncTimeout       = 5 * time.Minute
+	statusCheckInterval = 3 * time.Second
+)
+
+type pbmSeverity int
+
+const (
+	pbmFatal pbmSeverity = iota
+	pbmError
+	pbmWarning
+	pbmInfo
+	pbmDebug
+)
+
+func (s pbmSeverity) String() string {
+	switch s {
+	case pbmFatal:
+		return "F"
+	case pbmError:
+		return "E"
+	case pbmWarning:
+		return "W"
+	case pbmInfo:
+		return "I"
+	case pbmDebug:
+		return "D"
+	default:
+		return ""
+	}
+}
+
+type pbmLogEntry struct {
+	TS         int64 `json:"ts"`
+	pbmLogKeys `json:",inline"`
+	Msg        string `json:"msg"`
+}
+
+func (e pbmLogEntry) String() string {
+	return fmt.Sprintf("%s %s [%s/%s] [%s/%s] %s",
+		time.Unix(e.TS, 0).Format(time.RFC3339), e.Severity, e.RS, e.Node, e.Event, e.ObjName, e.Msg)
+}
+
+type pbmLogKeys struct {
+	Severity pbmSeverity `json:"s"`
+	RS       string      `json:"rs"`
+	Node     string      `json:"node"`
+	Event    string      `json:"e"`
+	ObjName  string      `json:"eobj"`
+	OPID     string      `json:"opid,omitempty"`
+}
+
+type pbmBackup struct {
+	Name    string `json:"name"`
+	Storage string `json:"storage"`
+}
+
+type pbmRestore struct {
+	Snapshot string `json:"snapshot"`
+}
+
+type pbmSnapshot struct {
+	Name       string `json:"name"`
+	Status     string `json:"status"`
+	Error      string `json:"error"`
+	CompleteTS int    `json:"completeTS"`
+	PbmVersion string `json:"pbmVersion"`
+}
+
+type pbmList struct {
+	Snapshots []pbmSnapshot `json:"snapshots"`
+	Pitr      struct {
+		On     bool        `json:"on"`
+		Ranges interface{} `json:"ranges"`
+	} `json:"pitr"`
+}
+
+type pbmListRestore struct {
+	Start    int    `json:"start"`
+	Status   string `json:"status"`
+	Type     string `json:"type"`
+	Snapshot string `json:"snapshot"`
+	Name     string `json:"name"`
+	Error    string `json:"error"`
+}
+
+type pbmStatus struct {
+	Backups struct {
+		Type       string        `json:"type"`
+		Path       string        `json:"path"`
+		Region     string        `json:"region"`
+		Snapshot   []pbmSnapshot `json:"snapshot"`
+		PitrChunks struct {
+			Size int `json:"size"`
+		} `json:"pitrChunks"`
+	} `json:"backups"`
+	Cluster []struct {
+		Rs    string `json:"rs"`
+		Nodes []struct {
+			Host  string `json:"host"`
+			Agent string `json:"agent"`
+			Ok    bool   `json:"ok"`
+		} `json:"nodes"`
+	} `json:"cluster"`
+	Pitr struct {
+		Conf bool `json:"conf"`
+		Run  bool `json:"run"`
+	} `json:"pitr"`
+	Running struct {
+		Type    string `json:"type"`
+		Name    string `json:"name"`
+		StartTS int    `json:"startTS"`
+		Status  string `json:"status"`
+		OpID    string `json:"opID"`
+	} `json:"running"`
+}
+
+func execPBMCommand(ctx context.Context, dbURL *url.URL, to interface{}, args ...string) error {
+	nCtx, cancel := context.WithTimeout(ctx, cmdTimeout)
+	defer cancel()
+
+	args = append(args, "--out=json", "--mongodb-uri="+dbURL.String())
+	cmd := exec.CommandContext(nCtx, pbmBin, args...) // #nosec G204
+
+	b, err := cmd.Output()
```

Comment:
maybe use CombinedOutput?

Reply:
It seems that there is no need for that here.  `ExitError` contains stderr.